src/app/application.cpp

```diff
@@ -403,15 +403,63 @@ void Application::processParams(const QStringList &params)
         return;
     }
 #endif
+    BitTorrent::AddTorrentParams torrentParams;
+    bool skipTorrentDialog = false;
 
     foreach (QString param, params) {
         param = param.trimmed();
+
+        // Process strings indicating options specified by the user. Use of
+        // any of these should prevent the "Add new torrent" dialog from
+        // opening:
+
+        if (param.startsWith("@savePath=")) {
+            torrentParams.savePath = param.mid(10);
+            skipTorrentDialog = true;
```

Comment:
Can't agree with you. IMO, it would be perfectly fine to supply a save path, a magnet uri and expect to see the dialog. And that is why I asked you to add the option to control it. It is perfectly fine to skip the dialog by default, but where is `skipTorrentDialog = false` then?